### ISSUE TITLE
RGW DNS names dont't work with individual and shared pools

### ISSUE THREAD

**Issue by CalmVibes** [rook/rook#13994]
Tested [rgw dns names](https://github.com/rook/rook/pull/13326) with the rook/ceph:master tag. This only works in one scenario - when using a multisite and specifying a zone for object store. When I don't specify a zone I get an error. Both in the case of specifying individual metadataPool and dataPool, and in the case of using shared pools.

Operator's log:

`2024-03-28 16:18:53.065957 E | ceph-object-controller: failed to reconcile CephObjectStore "rook-ceph/ceph-objectstore-88". failed to create object store deployments: failed to configure multisite for object store: failed create ceph multisite for object-store ["ceph-objectstore-88"]: failed to commit config changes after creating multisite config for CephObjectStore "rook-ceph/ceph-objectstore-88": failed to commit RGW configuration period changes%!!(MISSING)(EXTRA []string=[]): exit status 22`

More details:
[operator.log](https://github.com/rook/rook/files/14792680/operator.log)

ceph-objectstore-88 with shared pools, ceph-objectstore-89 is with individual metadataPool and dataPool

Let me know if I can provide any other information

**Comment by thotz** [comment]
@CalmVibes 
can u please tell me when the feature is working?
I am guessing for normal object stores it works.

If u are using multisite it is expected to give `zone` name in objectstore CRD.

for shared pools, i am not sure whether it is tested with multisite well. can u please check shared pools with non multisite config for this feature?

**Comment by CalmVibes** [comment]
> can u please check shared pools with non multisite config for this feature?

Yes, ceph-objectstore-88 and ceph-objectstore-89 object stores in this issue were created after setting up the multisite zone, realm, zone group.

Now I created object stores in clear Rook/Ceph cluster. And RGW DNS names work fine. Checked for individual object stores and object stores with shared pools. 

The only point I noticed is that when specifying object store parameters with hosting.dnsNames value directly in the [helm's chart values.yaml](https://github.com/rook/rook/blob/master/deploy/charts/rook-ceph-cluster/values.yaml), the store is created without a hosting parameter.

And this moment about CRD https://github.com/rook/rook/pull/13326#issuecomment-2025631645

**Comment by travisn** [comment]
> Now I created object stores in clear Rook/Ceph cluster. And RGW DNS names work fine. Checked for individual object stores and object stores with shared pools.

Good to hear it's working

> The only point I noticed is that when specifying object store parameters with hosting.dnsNames value directly in the [helm's chart values.yaml](https://github.com/rook/rook/blob/master/deploy/charts/rook-ceph-cluster/values.yaml), the store is created without a hosting parameter.

So you're saying it does **NOT** work when specifying the dnsNames in the values.yaml? Can you share the following?
1. the snippet from your values.yaml
2. the CephObjectStore yaml that results after you install? (`kubectl -n rook-ceph get cephobjectstores -o yaml) 
3. the rook operator log

**Comment by travisn** [comment]
@CalmVibes The `hosting` section is in the wrong place. It needs to go under the `cephObjectStores.spec` section of the values.yaml. It is currently found under the `storageClass` section, so it is ignored.

**Comment by CalmVibes** [comment]
> @CalmVibes The `hosting` section is in the wrong place. It needs to go under the `cephObjectStores.spec` section of the values.yaml. It is currently found under the `storageClass` section, so it is ignored.

That's right, I was mistakenly convinced that `hosting` was still part of the `cephObjectStores.spec` section. After adjustments, RGW DNS names work correctly. 

Thank you for this functionality improvement!
